pkg/scheduler/core/generic_scheduler.go

```diff
@@ -51,22 +51,55 @@ func (g *genericScheduler) Schedule(ctx context.Context, placement *policyv1alph
 		return result, fmt.Errorf("no clusters available to schedule")
 	}
 
-	feasibleClusters, err := g.findClustersThatFit(ctx, g.scheduleFramework, placement, &spec.Resource, clusterInfoSnapshot)
+	// 1. calculate reserved clusters and candidate clusters
+	// 2. prioritize candidate clusters
+	// 3. select target clusters from (reserved clusters + candidate clusters)
+	// 4. assign replicas
+	readyClusters, unreadyClusters, unjoinedClusters := clusterInfoSnapshot.InspectClusters()
+	bindClusters := util.ConvertToClusterNames(spec.Clusters)
+	reservedClusters := calcReservedCluster(bindClusters, readyClusters)
+	availableClusters := calcAvailableCluster(bindClusters, readyClusters)
+
+	candidateClusters, err := g.findClustersThatFit(ctx, g.scheduleFramework, placement, &spec.Resource, clusterInfoSnapshot, availableClusters)
 	if err != nil {
 		return result, fmt.Errorf("failed to findClustersThatFit: %v", err)
 	}
-	if len(feasibleClusters) == 0 {
-		return result, fmt.Errorf("no clusters fit")
+
+	if len(candidateClusters) == 0 {
+		// just warn
+		klog.Warningf("There is no candidate clusters")
+	} else {
+		klog.V(4).Infof("Candidate clusters found: %v", candidateClusters)
+	}
+
+	if unreadyClusters.Len() > 0 {
+		if !features.FeatureGate.Enabled(features.Failover) {
+			// reserve unready clusters if failover is disabled
+			reservedClusters = reservedClusters.Union(unreadyClusters)
```

Comment:
It seems like not correct.
If origin schedule result is [A, B, C], all clusters joined are [A, B, C, D, E], and [C, D] are not ready.
So, the reservedClusters is [A, B], unreadyClusters is [C, D], the union result is [A, B, C, D]? Correct me if I misunderstanding it.

Reply:
Oh yes, sorry I will fix it soon